csharp/src/Google.Protobuf.Test/WellKnownTypes/WrappersTest.cs

```diff
@@ -71,8 +71,7 @@ public void NonDefaultSingleValues()
                 Uint64Field = 4
             };
 
-            var bytes = message.ToByteArray();
-            var parsed = TestWellKnownTypes.Parser.ParseFrom(bytes);
+            var parsed = MessageParsingHelpers.AssertRoundtrip(TestWellKnownTypes.Parser, message);
```

Comment:
alternatively, we could parametrize the test  with a `bool useBufferSerialization`.
This test doesn't do exactly what the original version (the method always returns message parsed by the legacy ParseFrom() )

Reply:
Method now takes an action that is called for each parsed message